src/Sentry.Unity/Integrations/TraceGenerationIntegration.cs

```diff
@@ -0,0 +1,39 @@
+using Sentry.Extensibility;
+using Sentry.Integrations;
+
+namespace Sentry.Unity.Integrations;
+
+internal sealed class TraceGenerationIntegration : ISdkIntegration
+{
+    private readonly ISceneManager _sceneManager;
+    private readonly ISentryMonoBehaviour _sentryMonoBehaviour;
+
+    public TraceGenerationIntegration(SentryMonoBehaviour sentryMonoBehaviour) : this(sentryMonoBehaviour, SceneManagerAdapter.Instance)
+    { }
+
+    internal TraceGenerationIntegration(ISentryMonoBehaviour sentryMonoBehaviour, ISceneManager sceneManager)
+    {
+        _sceneManager = sceneManager;
+        _sentryMonoBehaviour = sentryMonoBehaviour;
+    }
+
+    public void Register(IHub hub, SentryOptions options)
+    {
+        hub.ConfigureScope(scope =>
+            scope.SetPropagationContext(new SentryPropagationContext(SentryId.Create(), SpanId.Create())));
```

Comment:
This constructor feels like a lot of boiler plate. In fact even having to pass a new propagation context feels redundant as it seems it's the main use case we have, just setting a new propagation context.

Could also be solved with an extension method like `ResetPropagationContext()` that calls `SetPropagationContext(new SentryPropagationContext(SentryId.Create(), SpanId.Create())))`